code/drasil-lang/lib/Drasil/Database/UID.hs

```diff
@@ -23,51 +23,42 @@ class HasUID c where
   -- | Provides a /unique/ id for internal Drasil use.
   uid :: Getter c UID
 
--- | A @UID@ is a 'unique identifier' for things that we will put into our database
--- of information. We use a newtype wrapper to make sure we are only using
--- 'UID's where desired.
+-- | A @UID@ is a 'unique identifier' for things that we will put into our
+-- database of information. We use a newtype wrapper to make sure we are only
+-- using 'UID's where desired.
 data UID = UID { _namespace :: [String], _baseName :: String }
-  deriving (Eq, Ord, Generic) --, ToJSONKey)
+  deriving (Eq, Ord, Generic)
 
 makeLenses ''UID
 
-fullName :: UID -> [String]
-fullName u = u ^. namespace ++ [u ^. baseName]
-
 instance ToJSON UID where
-  toJSON = toJSON . fullName
+  toJSON = toJSON . show
 
 instance ToJSONKey UID where
   toJSONKey = toJSONKeyText (pack . show)
 
 instance Show UID where
-  show = intercalate ":" . fullName
+  show u = intercalate ":" $ u ^. baseName : u ^. namespace
 
--- | Smart constructor for making a 'UID' from a 'String'.
+-- | Smart constructor for 'UID's from raw 'String's.
 mkUid :: String -> UID
 mkUid s = UID { _namespace = [], _baseName = s }
-  -- '►' `elem` s = error $ "► not allowed in UID " ++ show s -- FIXME: Need to implement other constructors before we can use this.
-  -- null s       = error "UID must be non-zero length" -- FIXME: See Drasil.DocumentLanguage.TraceabilityGraph (uses an empty UID)
-  -- otherwise    = UID s
 
--- | Nest UID under a namespace
+-- | Nest a 'UID' under a namespace.
 nsUid :: String -> UID -> UID
 nsUid ns = over namespace (ns:)
```

Comment:
This nesting structure is the reason why I moved the basename to the front. Consider the case where we nest something $n$ times. If it starts in `A`, and then we nest it in `B` within `A`, the list is reversed, so appending the final UID to the end is confusing. e.g., `[D,C,B,A]` means we first nested in `A` (the "largest" thing), and then nested the concept within `B`, etc. So, if a chunk has title $T$, we want to see $T:D:C:B:A$.

This might not have been strictly necessary, in part because I think we're going to end up redoing `UID`s entirely, but I _think_ this implementation makes more sense.

Reply:
Agree with this choice.